src/gui/properties/propertieswidget.cpp

```diff
@@ -536,12 +522,38 @@ void PropertiesWidget::loadDynamicData()
         {
             qDebug("Updating priorities in files tab");
             m_ui->filesList->setUpdatesEnabled(false);
-            m_propListModel->model()->updateFilesProgress(m_torrent->filesProgress());
-            m_propListModel->model()->updateFilesAvailability(m_torrent->availableFileFractions());
-            // XXX: We don't update file priorities regularly for performance
-            // reasons. This means that priorities will not be updated if
-            // set from the Web UI.
-            // PropListModel->model()->updateFilesPriorities(h.file_priorities());
+
+            // Load torrent content if not yet done so
+            const bool isContentInitialized = m_propListModel->model()->hasIndex(0, 0);
+            if (!isContentInitialized)
+            {
+                // List files in torrent
+                m_propListModel->model()->setupModelData(m_torrent->info());
+                // Load file priorities
+                m_propListModel->model()->updateFilesPriorities(m_torrent->filePriorities());
+                // Update file progress/availability
+                m_propListModel->model()->updateFilesProgress(m_torrent->filesProgress());
+                m_propListModel->model()->updateFilesAvailability(m_torrent->availableFileFractions());
+
+                // Expand single-item folders recursively.
+                // This will trigger sorting and filtering so do it after all relevant data is loaded.
+                QModelIndex currentIndex;
+                while (m_propListModel->rowCount(currentIndex) == 1)
+                {
+                    currentIndex = m_propListModel->index(0, 0, currentIndex);
+                    m_ui->filesList->setExpanded(currentIndex, true);
+                }
+            } else {
```

Comment:
broken coding style here, please fix it.
```c++
}
else
{
```

Reply:
Fixed. Sorry for that, I'm still getting used to the coding style. I tried `uncrustify.cfg` but it's broken i.e. an opening curly bracket not on a new line...